pkg/image/importer/credentials.go

```diff
@@ -0,0 +1,112 @@
+package importer
+
+import (
+	"fmt"
+	"net/url"
+	"sync"
+
+	"github.com/golang/glog"
+
+	"github.com/docker/distribution/registry/client/auth"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/credentialprovider"
+	"k8s.io/kubernetes/pkg/util"
+)
+
+var (
+	NoCredentials auth.CredentialStore = &noopCredentialStore{}
+
+	emptyKeyring = &credentialprovider.BasicDockerKeyring{}
+)
+
+type noopCredentialStore struct{}
+
+func (s *noopCredentialStore) Basic(url *url.URL) (string, string) {
```

Comment:
I'm assuming this is for `docker/distribution/registry/client/auth.CredentialStore`, but that's not a commonly known one.  Type assertions?